pkg/reporter/json_report.go

```diff
@@ -257,6 +263,15 @@ func (j *jsonReportGenerator) buildJsonPackageReportFromPackage(p *models.Packag
 			Aliases:    utils.SafelyGetValue(vuln.Aliases),
 			Severities: severties,
 		})
+
+		if (len(pkg.Vulnerabilities) > 0 && flag) {
```

Comment:
Not sure what is the purpose of `flag` here. Can you please clarify?

Reply:
It's a boolean variable that simply runs the if block once. I added it cause there were scenarios where a package had multiple vulns, then the loop would add `target_alternate_package_version` multiple times which is not required.